File: projet_groupe4/src/main/java/projet_groupe4/model/TableJeu.java
package projet_groupe4.model;

import java.util.ArrayList;
import java.util.List;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;

@Entity
@Table(name = "table_jeu")
public class TableJeu {
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)

	private Integer id;
	@Column(name = "nom_table", columnDefinition = "varchar(50)", nullable = false)

	private String nomTable;
	@Column(nullable = false)

	private int capacite;
	@OneToMany(mappedBy = "tableJeu")
	private List<Reservation> reservations = new ArrayList<Reservation>();
	@Column(name="img_url", columnDefinition="varchar(100)")
	private String imgUrl;
	// Constructors
	public TableJeu() {
	}

	public TableJeu(Integer id, String nomTable, int capacite) {
		this.id = id;
		this.nomTable = nomTable;
		this.capacite = capacite;
	}

	public TableJeu(String nomTable, int capacite) {
		this.nomTable = nomTable;
		this.capacite = capacite;
	}

	// Getters & Setters
	public Integer getId() {
		return id;
	}

	public void setId(Integer id) {
		this.id = id;
	}

	public String getNomTable() {
		return nomTable;
	}

	public void setNomTable(String nomTable) {
		this.nomTable = nomTable;
	}

	public int getCapacite() {
		return capacite;
	}

	public void setCapacite(int capacite) {
		this.capacite = capacite;
	}
	public String getImgUrl() { return imgUrl; }
	public void setImgUrl(String imgUrl) { this.imgUrl = imgUrl; }

	// toString
	@Override
	public String toString() {
		return "Table [id=" + id + ", nomTable=" + nomTable + ", capacite=" + capacite + "]";
	}
	
	

}

File: projet_groupe4_angular/src/app/dto/table-jeu.ts
import { Reservation } from "./reservation";

export class TableJeu {


    constructor(
    public _id: number,
    public _nomTable: string,
    public _capacite: number,
    public _imgUrl : string,
    public _reservations?: Reservation[]
    
  ) {}

  // Getters & Setters
  public get id(): number {
    return this._id;
  }
  public set id(value: number) {
    this._id = value;
  }

  public get nomTable(): string {
    return this._nomTable;
  }
  public set nomTable(value: string) {
    this._nomTable = value;
  }

  public get capacite(): number {
    return this._capacite;
  }
  public set capacite(value: number) {
    this._capacite = value;
  }



  public get reservations(): Reservation[]| undefined {
    return this._reservations;
  }
  public set reservations(value: Reservation[]| undefined) {
    this._reservations = value;
  }

    public get imgUrl(): string {
    return this._imgUrl;
  }
  public set imgUrl(value: string) {
    this._imgUrl = value;
  }

  public toJson(): any {
    return {
      nomTable: this._nomTable,
      capacite: this._capacite,
      reservations: this._reservations?.map(r => r.toJson())|| [],
      imgUrl:this._imgUrl
    };
  }
}
